portal/server/controllers/utilisation-report-service/record-correction/record-correction-history/helpers.test.ts

```diff
@@ -0,0 +1,161 @@
+          formattedBankCommentary: firstCompletedCorrection.bankCommentary,
+        },
+        {
+          dateSent: {
+            formattedDateSent: '17 Mar 2024',
+            dataSortValue: 1,
+          },
+          exporter: secondCompletedCorrection.exporter,
+          formattedReasons: secondCompletedCorrection.formattedReasons,
+          formattedPreviousValues: secondCompletedCorrection.formattedPreviousValues,
+          formattedCorrectedValues: secondCompletedCorrection.formattedCorrectedValues,
+          formattedBankCommentary: '-',
+        },
+      ];
+
+      expect(viewModel).toEqual(expectedViewModel);
+    });
+
+    it('should map the completed correction "date sent" to a formatted date sent sorted by date ascending', () => {
+      // Arrange
+      const aCompletedCorrection = () => ({
+        id: 1,
+        dateSent: new Date('2024-02-01').toISOString(),
+        exporter: 'An exporter',
+        formattedReasons: 'Other',
+        formattedPreviousValues: '-',
+        formattedCorrectedValues: '-',
+        formattedBankCommentary: 'Some bank commentary',
+      });
+
+      const completedCorrections: GetCompletedFeeRecordCorrectionsResponseBody = [
+        {
+          ...aCompletedCorrection(),
+          id: 1,
+          dateSent: new Date('2024-06-01').toISOString(), // '01 Jun 2024', dataSortValue = 3
+        },
+        {
+          ...aCompletedCorrection(),
+          id: 2,
+          dateSent: new Date('2024-07-28').toISOString(), // '28 Jul 2024', dataSortValue = 5
+        },
+        {
+          ...aCompletedCorrection(),
+          id: 3,
+          dateSent: new Date('2024-03-14').toISOString(), // '14 Mar 2024', dataSortValue = 0
+        },
+        {
+          ...aCompletedCorrection(),
+          id: 4,
+          dateSent: new Date('2024-05-07').toISOString(), // '07 May 2024', dataSortValue = 2
+        },
+        {
+          ...aCompletedCorrection(),
+          id: 5,
+          dateSent: new Date('2024-04-01').toISOString(), // '01 Apr 2024', dataSortValue = 1
+        },
+        {
+          ...aCompletedCorrection(),
+          id: 6,
+          dateSent: new Date('2024-06-21').toISOString(), // '21 Jun 2024', dataSortValue = 4
+        },
+      ];
+
+      // Act
+      const viewModel = mapCompletedCorrectionsToViewModel(completedCorrections);
+
+      // Assert
+      expect(viewModel).toHaveLength(6);
+
+      const expectedMappedDateSentValues = [
+        { formattedDateSent: '01 Jun 2024', dataSortValue: 3 },
+        { formattedDateSent: '28 Jul 2024', dataSortValue: 5 },
+        { formattedDateSent: '14 Mar 2024', dataSortValue: 0 },
+        { formattedDateSent: '07 May 2024', dataSortValue: 2 },
+        { formattedDateSent: '01 Apr 2024', dataSortValue: 1 },
+        { formattedDateSent: '21 Jun 2024', dataSortValue: 4 },
+      ];
+
+      expect(viewModel.map(({ dateSent }) => dateSent)).toEqual(expectedMappedDateSentValues);
```

Comment:
Personally I would write this:
```suggestion
      viewModel.forEach((row, index) => {
      		expect(row.dateSent).toEqual(expectedDateSentValues[index]);
      });
```
I think I find it easier to comprehend but might just be me.

And tbh I actually might have just written it like this:
```
expect(viewModel[0].dateSent).toEqual({ formattedDateSent: '01 Jun 2024', dataSortValue: 3 });
expect(viewModel[1].dateSent).toEqual({ formattedDateSent: '28 Jul 2024', dataSortValue: 5 });
...
```
It's actually less lines of code.